.github/workflows/build-and-tag-maven.yml

```diff
@@ -0,0 +1,100 @@
+name: Build and tag Maven project
+
+on:
+  workflow_call:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Alfresco/alfresco-build-tools/.github/actions/pre-commit@v1.5.0
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    needs: pre-commit
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@860f60056505705214d223b91ed7a30f173f6142
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: 'maven'
+
+      - name: Update pom files to the new version
+        id: update-pom-to-next-version
+        if: ${{ github.event_name == 'push' }}
+        uses: Alfresco/alfresco-build-tools/.github/actions/update-pom-to-next-pre-release@aae-8964-produce-alphas
+
+      - name: Set version env variable
+        if: ${{ github.event_name == 'push' }}
```

Comment:
I wonder if all the checks on `${{ github.event_name == 'push' }}` should be replaced by a check on an input boolean, so that the caller can decide in which cases the push should be done, wdyt?

Reply:
It makes sense, but I think it's better to do it when we know exactly what we want. For instance, we may need different flags (`pushDocker`, `pushMaven`, `createTag`). For the moment it would only mean an extra parameter to pass while calling the workflow that will have always the same value.

Reply:
As you prefer :+1: